Library/Homebrew/cmd/search.rb

```diff
@@ -101,21 +101,15 @@ def query_regexp(query)
   end
 
   def search_taps(query)
-    valid_dirnames = ["Formula", "HomebrewFormula", "Casks", ".", ""].freeze
-    q = "user:Homebrew%20user:caskroom%20filename:#{query}"
-    GitHub.open "https://api.github.com/search/code?q=#{q}" do |json|
-      json["items"].map do |object|
-        dirname, filename = File.split(object["path"])
-        next unless valid_dirnames.include?(dirname)
-        user = object["repository"]["owner"]["login"]
-        user = user.downcase if user == "Homebrew"
-        repo = object["repository"]["name"].sub(/^homebrew-/, "")
-        tap = Tap.fetch user, repo
-        next if tap.installed?
-        basename = File.basename(filename, ".rb")
-        "#{user}/#{repo}/#{basename}"
-      end.compact
-    end
+    valid_dirnames = ["Formula", "HomebrewFormula", "Casks", "."].freeze
```

Comment:
Are you sure you can drop `""` here? Have you verified this with formulae in the root e.g. homebrew/science?